air/src/stack/mod.rs

```diff
@@ -1,3 +1,70 @@
-use super::EvaluationFrame;
+use super::{EvaluationFrame, Felt, FieldElement, TransitionConstraintDegree, Vec};
 
+mod field_ops;
 pub mod op_flags;
+mod stack_manipulation;
+mod system_ops;
+
+// STACK OPERATIONS TRANSITION CONSTRAINTS
+// ================================================================================================
+
+/// Builds the transition constraint degrees for the stack module and all stack operation components.
+pub fn get_transition_constraint_degrees() -> Vec<TransitionConstraintDegree> {
+    // stack manipulation operations contraints degrees.
+    let mut degrees = stack_manipulation::get_transition_constraint_degrees();
+    // system operations contraints degrees.
+    degrees.append(&mut system_ops::get_transition_constraint_degrees());
+    // field operations contraints degrees.
+    degrees.append(&mut field_ops::get_transition_constraint_degrees());
+
+    degrees
+}
+
+/// Returns the number of transition constraints for the stack operations.
+pub fn get_transition_constraint_count() -> usize {
+    stack_manipulation::get_transition_constraint_count()
+        + system_ops::get_transition_constraint_count()
+        + field_ops::get_transition_constraint_count()
+}
+
+/// Enforces constraints for the stack module and all stack operations.
+pub fn enforce_constraints<E: FieldElement<BaseField = Felt>>(
+    frame: &EvaluationFrame<E>,
+    result: &mut [E],
+) -> usize {
+    let mut index = 0;
+
+    let frame_copy = frame.clone();
+
+    let op_flag = op_flags::OpFlags::new(frame_copy);
```

Comment:
We should try to avoid cloning here as it could result in a lot of extra overhead. This should be simple as I don't think `OpFlags::new()` actually needs to take ownership of the frame.

Also, building `op_flag` here is fine for now, but in the future we'll probably need to move it one level higher as we'll need to have access to op flags in the decoder as well.

Reply:
Makes sense. In the earlier PR, OpFlag was consuming the frame. I'm passing a reference now of the frame so that it doesn't get consume.